test/test/cpu/CpuTests.java

```diff
@@ -32,4 +34,36 @@ public void itimerTotal(TestProcess p) throws Exception {
         Output out = p.profile("-d 2 -e itimer -i 100ms --total -o collapsed");
         assertCloseTo(out.total(), 2_000_000_000, "itimer total should match profiling duration");
     }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX, cpu = 1)
+    public void perfEventsWrongTargetCpu(TestProcess p) throws Exception {
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 2");
+        assertCloseTo(out.total(), 0, "perf_events total should be 0 when the wrong CPU is targeted");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX, cpu = 1)
+    public void perfEventsRightTargetCpu(TestProcess p) throws Exception {
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 1");
+        assertCloseTo(out.total(), 2_000_000_000, "perf_events total should match profiling duration");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void itimerDoesNotSupportTargetCpu(TestProcess p) throws Exception {
+        try {
+            Output out = p.profile("-e itimer --target-cpu 1");
+            throw new IllegalStateException("Profiling should have failed");
+        } catch (IOException expectedException) {
```

Comment:
Do we have a cleaner way to test that an exception has been thrown? I'm used to JUnit 5 (e.g. `assertThrows)

Reply:
No, but you can create one if you wish (preferably in a separate PR).